src/sqlfluff/core/rules/base.py

```diff
@@ -416,6 +435,59 @@ def raw_segments(self):
         )
 
 
+class CrawlBehavior:
+    """Implements how a lint rule traverses the parse tree."""
+
+    def __init__(
+        self, works_on_unparsable: bool, recurse_into: bool, needs_raw_stack: bool
+    ):
+        self.works_on_unparsable = works_on_unparsable
+        self.recurse_into = recurse_into
+        self.raw_stack: Optional[List[RawSegment]]
+        if needs_raw_stack:
+            self.raw_stack = []
+        else:
+            self.raw_stack = None  # pragma: no cover
```

Comment:
Can you not drop the else and assign directly (or with a ternary) - seems cleaner and avoids the no cover
```suggestion
        self.raw_stack: Optional[List[RawSegment]] = [] if needs_raw_stack else None
```

Reply:
Will do! I was debating with myself earlier the best way to do this.

Reply:
I'll do this in the other PR, to avoid having to wait for another CI cycle.

Reply:
Welp, I'll go ahead and do it now. Got to "Update branch" from `main` anyway.